rancher2/resource_rancher2_token.go

```diff
@@ -117,3 +117,24 @@ func resourceRancher2TokenDelete(d *schema.ResourceData, meta interface{}) error
 	d.SetId("")
 	return nil
 }
+
+func isTokenValid(c *Config, id string) (bool, error) {
+	if len(id) == 0 {
+		return false, nil
+	}
```

Comment:
Wouldn't a kubeconfig with CLI commands instead of a token fail this? There is a setting `kubeconfig-generate-token` that when enabled uses the cli to generate a token. Instead of embedding it into the kubeconfig it just lets the CLI handle it each time, CLI does cache so that won't be a problem. However, I think this may interpret that setup as not having a valid token.

Reply:
Nope, i've tested the user case and is working fine. The token is replaced just if not empty https://github.com/rancher/terraform-provider-rancher2/pull/837/files#diff-035d1bb527d3ce1fde79a8bf6a191e1c441f7e2074f17caa106ba35a66fefa0bR697 , when `kubeconfig-generate-token=false` the `generateKubeconfig` action is not generating token nor diff at kube_config